### PR TITLE
NIAID-236

### PR DESCRIPTION
**Pull request by buske**

### CODE REVIEW

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientClass.xml [hunk 1]
```diff
@@ -29,8 +29,8 @@
   <parent>XWiki.XWikiClasses</parent>
   <author>xwiki:XWiki.Admin</author>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
-  <date>1426869717000</date>
-  <contentUpdateDate>1426869717000</contentUpdateDate>
+  <date>1559061018000</date>
+  <contentUpdateDate>1559061018000</contentUpdateDate>
```
Comment (danielpgross) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientClass.xml]: In the interest of keeping the commitdiff clean, date changes shouldn't be included.

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml [hunk 2]
```diff
@@ -29,8 +29,8 @@
   <parent>PhenoTips.PatientClass</parent>
   <author>xwiki:XWiki.Admin</author>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
-  <date>1401822211000</date>
-  <contentUpdateDate>1401822211000</contentUpdateDate>
+  <date>1559057968000</date>
+  <contentUpdateDate>1558991504000</contentUpdateDate>
```
Comment (danielpgross) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: Remove

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__clinical_diagnosis.xml [hunk 3]
```diff
@@ -29,8 +29,8 @@
   <parent>PhenoTips.PatientSheet</parent>
   <author>xwiki:XWiki.Admin</author>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
-  <date>1488315718000</date>
-  <contentUpdateDate>1488315713000</contentUpdateDate>
+  <date>1559062361000</date>
+  <contentUpdateDate>1558991504000</contentUpdateDate>
```
Comment (danielpgross) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__clinical_diagnosis.xml]: Remove

File: components/pedigree/resources/src/main/resources/pedigree/externalEndpoints.js [hunk 4]
```diff
@@ -140,6 +140,15 @@ define([
             return new XWiki.Document('WebHome', 'data').getRestURL('objects/PhenoTips.PedigreeClass/');
         },
 
+        getDisorderDetailsURL: function(term) {
+            // Must be a term of the form: PREFIX:XXXXXX
+            return XWiki.contextPath + "/rest/vocabularies/terms/" + term;
+        },
+
+        getDisorderSuggestURL: function(input) {
```
Comment (danielpgross) [inline components/pedigree/resources/src/main/resources/pedigree/externalEndpoints.js]: Why does this take the `input` argument if it's never used?

File: components/pedigree/resources/src/main/resources/pedigree/disorder.js [hunk 5]
```diff
@@ -40,31 +35,49 @@ define([
...
-            var queryURL           = baseOMIMServiceURL + "&q=id:" + this._disorderID;
-            //console.log("queryURL: " + queryURL);
-            new Ajax.Request(queryURL, {
-                method: "GET",
-                onSuccess: this.onDataReady.bind(this),
-                //onComplete: complete.bind(this)
-                onComplete: callWhenReady ? callWhenReady : {}
-            });
+            // Convert to fully-prefixed term
+            var curie = this._disorderID;
```
Comment (danielpgross) [inline components/pedigree/resources/src/main/resources/pedigree/disorder.js:39]: [curie](https://www.dictionary.com/browse/curie) is a unit of radioactivity, why is this the variable name?

File: components/vocabularies/mondo/api/src/main/java/org/phenotips/vocabulary/internal/solr/MondoDiseaseOntology.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+ * Provides access to the Mondo Disease Ontology (MONDO). The vocabulary prefix is {@code MONDO}.
+ *
+ * @version $Id$
+ */
+@Component
+@Named("mondo")
+@Singleton
+public class MondoDiseaseOntology extends AbstractOBOSolrVocabulary
+{
+    /** For determining if a query is a an id. */
```
Comment (danielpgross) [inline components/vocabularies/mondo/api/src/main/java/org/phenotips/vocabulary/internal/solr/MondoDiseaseOntology.java]: Typo `a an`

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml [hunk 7]
```diff
@@ -29,8 +29,8 @@
   <parent>PhenoTips.WebHome</parent>
   <author>xwiki:XWiki.Admin</author>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
-  <date>1401822215000</date>
-  <contentUpdateDate>1401822215000</contentUpdateDate>
+  <date>1559060904000</date>
+  <contentUpdateDate>1558991510000</contentUpdateDate>
```
Comment (danielpgross) [inline components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: Remove

**Review by danielpgross — CHANGES_REQUESTED** [review]
Overall this is looking good 👍 

I did basic testing on a new patient and also on an imported patient with ORDO and OMIM terms, to test backwards-compatibility.

On the new patient:
All testing worked fine, there were no bugs or unexpected behaviour to note.

On the imported patient:
- ORDO terms were not shown at all on the patient form or in the pedigree editor
- OMIM terms worked fine on both the patient form and in the pedigree editor

A few notes on my review:
- I did not look in detail at the logic changes to the pedigree editor
- I assumed that all of the vocabulary changes (under `/components/vocabularies/mondo`) were copy/paste and did not look at them in detail

File: components/pedigree/resources/src/main/resources/pedigree/disorder.js (hunk 5 above)
Comment (buske) [reply components/pedigree/resources/src/main/resources/pedigree/disorder.js:39]: https://www.w3.org/TR/curie/

There's a nice summary here: https://schemablocks.org/formats/identifiers-curies/
> CURIEs (“Compact URIs”) are namespace-scoped identifiers which can be expanded to Internationalized Resource Identifiers (IRI). A CURIE is comprised of two components, a prefix and a reference, separated by a colon symbol (:). CURIES are case sensitive, although for prefixes this practice is inconsistently being followed.

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientClass.xml (hunk 1 above)
Comment (buske) [reply components/patient-data/ui/src/main/resources/PhenoTips/PatientClass.xml]: I usually would, but I thought we'd changed this policy as per the issues with the distro wizard that @sebialex encountered...

File: components/pedigree/resources/src/main/resources/pedigree/externalEndpoints.js (hunk 4 above)
Comment (buske) [reply components/pedigree/resources/src/main/resources/pedigree/externalEndpoints.js]: Good catch, that's definitely a relic from a refactor.

**Comment by buske** [comment]
> On the imported patient:
> * ORDO terms were not shown at all on the patient form or in the pedigree editor
> * OMIM terms worked fine on both the patient form and in the pedigree editor

👍 This is because the PR extends the behaviour of the OMIM field and the ORDO field is hidden. This is sufficient to meet the client reqs.

**Comment by buske** [comment]
>     * I did not look in detail at the logic changes to the pedigree editor

@danielpgross It would be useful to get your review on the Affected status decoupling commit: https://github.com/Gene42/phenotips/pull/5/commits/d0077d1402b1d39ffff2f10ea63ec0d2c5d5952e

File: components/vocabularies/mondo/api/src/main/java/org/phenotips/vocabulary/internal/solr/MondoDiseaseOntology.java (hunk 6 above)
Comment (buske) [reply components/vocabularies/mondo/api/src/main/java/org/phenotips/vocabulary/internal/solr/MondoDiseaseOntology.java]: Looks like this was a typo in the source file (HPO ontology) :(  Will fix.

File: components/pedigree/resources/src/main/resources/pedigree/disorder.js (hunk 5 above)
Comment (danielpgross) [reply components/pedigree/resources/src/main/resources/pedigree/disorder.js:39]: Good to know, thanks for explaining!

**Comment by danielpgross** [comment]
Thanks for addressing all the other comments in my last review.

Bug (as discussed in person):
1. Add 2 diagnoses to an individual in the pedigree.
2. Click "Delete all" under the "Diagnosis" field.
3. Only one diagnosis will be deleted and the following error will be logged in the console: `Uncaught TypeError: Cannot read property 'removeChild' of null`

**Comment by danielpgross** [comment]
Here's a behaviour that I don't think makes sense, although it's not a bug per se:
The `affected` non-standard disorder term is added when selecting "Affected" under carrier status. The `affected` non-standard disorder term is removed if "Not affected" is then selected.
BUT
The `affected` non-standard disorder term is _not_ removed if any other carrier status is then selected.

I think the `affected` non-standard disorder term should be removed if any carrier status other than "Affected" is selected.

**Comment by danielpgross** [comment]
Here's a behaviour that either does not make sense to me, or is a bug (i.e. I'm not sure if it was intentional). The behaviour was present before your changes, but this is a good time to make a decision about it.
1) With no diagnosis terms selected, change the carrier status to something other than "Affected", e.g. "Carrier".
2) Add a diagnosis term.
3) Remove the diagnosis term. The carrier status is reset to "Not affected".

Instead, I think the carrier status should not be changed when all diagnosis terms are removed.

**Comment by buske** [comment]
> Thanks for addressing all the other comments in my last review.
> 
> Bug (as discussed in person):
> 
>     1. Add 2 diagnoses to an individual in the pedigree.
> 
>     2. Click "Delete all" under the "Diagnosis" field.
> 
>     3. Only one diagnosis will be deleted and the following error will be logged in the console: `Uncaught TypeError: Cannot read property 'removeChild' of null`

After investigating further, this is a bug in 1.3.x unrelated to this PR and is fixed in 1.4.

**Comment by buske** [comment]
> Here's a behaviour that I don't think makes sense, although it's not a bug per se:
> The `affected` non-standard disorder term is added when selecting "Affected" under carrier status. The `affected` non-standard disorder term is removed if "Not affected" is then selected.
> BUT
> The `affected` non-standard disorder term is _not_ removed if any other carrier status is then selected.
> 
> I think the `affected` non-standard disorder term should be removed if any carrier status other than "Affected" is selected.

As discussed, this is probably the best of the available options for how this should behave.

**Comment by buske** [comment]
> Here's a behaviour that either does not make sense to me, or is a bug (i.e. I'm not sure if it was intentional). The behaviour was present before your changes, but this is a good time to make a decision about it.
> 
>     1. With no diagnosis terms selected, change the carrier status to something other than "Affected", e.g. "Carrier".
> 
>     2. Add a diagnosis term.
> 
>     3. Remove the diagnosis term. The carrier status is reset to "Not affected".
> 
> 
> Instead, I think the carrier status should not be changed when all diagnosis terms are removed.

Agree, this is a bug and vestigial behaviour. I've pushed a new commit that fixes this behaviour and simplifies some of the affected status logic.

**Review by danielpgross — APPROVED** [review]
Looks good 👍

**Comment by buske** [comment]
Closing, since this isn't intended to be merged into master.

**Pull request closed without merging**